packages/medusa/src/services/line-item.ts

```diff
@@ -332,6 +332,15 @@ class LineItemService extends TransactionBaseService {
       unit_price = context.variantPricing?.calculated_price ?? undefined
     }
 
+    if (unit_price == null) {
```

Comment:
This is a good fallback, but I find this too late in the process to identify that something that is purchasable doesn't have a price.

Reply:
Yeah it is really to prevent it to happen, but as discussed and published in eng core and as you mentioned, we should probably not be able to publish a product/variants that does not have any prices